### ISSUE TITLE
Please make it so that Denied is sent out when the X button is pressed in HostAccess Request.

### ISSUE THREAD

**Issue by kazu0617** [Neos-Metaverse/NeosPublic#2881]
### Is your feature request related to a problem? Please describe.

This is a problem when you are creating a program that prompts you according to the status of HTTP HostAccess acquisition, and if you delete the program with the X button, the process stops there and you cannot continue.

### Relevant issues

soft relevant. #904

### Describe the solution you'd like

Currently, in Neos, if you want to "reject this time", you can do so without being recorded in Web Hosts by clicking the X button. However, if the user presses the "X" button with that intention, LogiX will not be able to detect it and the process will stop there (even if some fallback process is written for the rejection).

For this reason, I would like to see the "X button" either disabled or pulsed as "Denied" when the X button is pressed.

### Describe alternatives you've considered

Elapsed Time to resend pulses at regular intervals until they are allowed or denied. We think this is very redundant and not a recommended method. Also, we do not understand the intention of "reject only once".

**Comment by kazu0617** [comment]
The reproduction itself should be easy to do if you build it like this.
![image](https://user-images.githubusercontent.com/6259214/131207007-512edc79-c6a3-4f10-a974-c145ebab5b3f.png)

An alternative is, for example, the following example.
![image](https://user-images.githubusercontent.com/6259214/131207149-9a9da95f-a3dd-4216-9f2f-205e013db056.png)

**Comment by SnooperXP** [comment]
I was wondering about this the other night in Neos when clicking the X that it acted much like an ignore, neither putting into the allow or deny. Would it be better if this is the expected case to make an ignore button and have that as a case, or should we keep it to just allow and deny? (Open question, just postulating)

**Comment by kazu0617** [comment]
If you send it to Denied, the existing LogiX will work as is, and I don't think this will be done often enough to add a new ignore, so my personal preference would be to send it to Denied, but I think you're right about that too.

original:
```
どちらも正解だとは思います。Deniedに送出してくれると、既存のLogiXはそのまま動作してくれますし、無視を新たに追加するほどこれが行われる回数が多いとは思わないので、自分個人の意見としてはDeniedに送信することを望みますが、その意見もその通りだと思います。
```

**Comment by AshtonSparx** [comment]
Or at least have a separate impulse for "Closed" or "Ignored" since they technically haven't denied it yet. just ignored the prompt

**Comment by ProbablePrime** [comment]
While Ashton is technically correct, Kazu is also correct.

We therefore have some options:
1. Do Nothing
2. Remove the Red X button from the Dialog. This forces a choice between Allow and Deny
3. Add the Ignored/Closed Impulse and for backward compatibility connect existing saved copies of this node to connect Ignored to the same path as denied. NEW uses of this node would not do this and could use ignored separately. This would only auto-connect old versions of the node.

**Comment by 3x1t-5tyl3** [comment]
Personally; I'd love to get a "ignored" output. It opens up further actions creators can take towards explaining why a request was opened.

**Comment by kazu0617** [comment]
When I think about it, it would be no problem to automatically connect to the old code. I'm also in favor of adding a new one, as well as the others.

**Comment by kazu0617** [comment]
> 
> 
> While Ashton is technically correct, Kazu is also correct.
> 
> We therefore have some options:
> 
>     1. Do Nothing
> 
>     2. Remove the Red X button from the Dialog. This forces a choice between Allow and Deny
> 
>     3. Add the Ignored/Closed Impulse and for backward compatibility connect existing saved copies of this node's new ignored output to the same path as denied. NEW uses of this node would not do this and could use ignored separately. This would only auto-connect old versions of the node.

I feel that option 3 is the most plausible, at least according to the opinions of the people participating in the Issue here (I was also struggling with it at first, but now I think 3 is the most plausible, and the rest of us chose that option from the beginning).
So, if you are not extremely busy, please, if you think you can implement @ProbablePrime, please do.

**Comment by ProbablePrime** [comment]
I started work on this today. It actually makes sense to do a 4th option here which more closely aligns with Ashton's comments.

I've added "OnIgnored" which is exclusively for the Red X button. It also does not persist this choice so that you can re-trigger the request. There are inbuilt protections against spamming the user here so don't trigger this too many times too quickly. 

Of course subject to review by the team but: https://i.probableprime.co.uk/210918-OkIS51S1bebJOpY7LlCf.mp4

**Comment by Frooxius** [comment]
Merged in 2021.9.20.1334! Thanks for report and implementation!
